starlette/responses.py

```diff
@@ -258,14 +259,15 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             except OSError:
                 raise ClientDisconnect()
         else:
-            async with anyio.create_task_group() as task_group:
+            with collapse_excgroups():
```

Comment:
Why was this added?

Reply:
it's subtle but it's needed to get test_custom_middleware to pass

the reason is app_exc shouldn't be subject to collapse_excgroups - eg it's from the user's code so might have use a TG or Nursery and therefore legitimately wrapped in a EG

however that means if something raises an exception from StreamingResponse they're not really expecting an optional background task to watch for disconnections, so they expect the EG to be unwrapped

Reply:
I mention that it's an optional background task in StreamingResponse, because it should not be the case that StreamingResponse raises a different kind of exception based on asgi version

Reply:
I don't want to change anything outside the base http middleware because of it

Reply:
we can nest the raise inside the collapse_excgroups() in base.py - but it's not really correct

Reply:
Can't we achieve correctness and not change the other part of the code? I would prefer to leave the other part untouched and make the base http middleware less correct rather than modify elsewhere because of it.

Reply:
I'll have a think

Reply:
Do you have an MRE to share of a code that would be a problem with the `raise app_exc` inside the `collapse_excgroups`?

The PR you are working in seems to also change the `StreamingResponse`, so it's not really avoiding changes in it.

Reply:
so the raise app_exc can't be put inside the async with of the task group because the app_exc might not be set until after the task group has finished, and you should not put anything in between a task group and an collapse_excgroups because then you're collapsing exc groups that you've not created. Thus it has to be this way.

If you want to avoid touching StreamingResponse in this PR, I've separated the changes out into another PR that's more like the old anyio 3 behaviour and strict_exception_groups=False

Reply:
There's a discussion ongoing in the trio gitter to make strict_exception_groups not deprecated and expose it in anyio

Reply:
> Do you have an MRE to share of a code that would be a problem with the `raise app_exc` inside the `collapse_excgroups`?
> 
> The PR you are working in seems to also change the `StreamingResponse`, so it's not really avoiding changes in it.

MRE converted to test cases and in #2830